packages/eslint-plugin/src/util/propertyTypes.ts

```diff
@@ -0,0 +1,36 @@
+import * as ts from 'typescript';
+
+export function getTypeOfPropertyOfName(
+  checker: ts.TypeChecker,
+  type: ts.Type,
+  name: string,
+  escapedName?: ts.__String,
+): ts.Type | undefined {
+  // Most names are directly usable in the checker and aren't different from escaped names
+  if (!escapedName || !name.startsWith('__')) {
+    return checker.getTypeOfPropertyOfType(type, name);
```

Comment:
This is the content of TypeScript's source for `getTypeOfPropertyOfType`:

```ts
// Return the type of the given property in the given type, or undefined if no such property exists
function getTypeOfPropertyOfType(type: Type, name: __String): Type | undefined {
    const prop = getPropertyOfType(type, name);
    return prop ? getTypeOfSymbol(prop) : undefined;
}
```

I can switch this file to use the same strategy if directed.